Sources/Moya/MoyaError.swift

```diff
@@ -26,39 +26,42 @@ public enum MoyaError: Swift.Error {
 
     /// Indicates that an `Endpoint` failed to map to a `URLRequest`.
     case requestMapping(String)
-
-    /// Indicates that an `Endpoint` failed to encode the parameters for the `URLRequest`.
-    case parameterEncoding(Swift.Error)
 }
 
 public extension MoyaError {
     /// Depending on error type, returns a `Response` object.
     var response: Moya.Response? {
         switch self {
-        case .imageMapping(let response): return response
-        case .jsonMapping(let response): return response
-        case .stringMapping(let response): return response
-        case .objectMapping(_, let response): return response
-        case .encodableMapping: return nil
-        case .statusCode(let response): return response
-        case .underlying(_, let response): return response
-        case .requestMapping: return nil
-        case .parameterEncoding: return nil
+        case let .imageMapping(response),
+             let .jsonMapping(response),
+             let .stringMapping(response),
+             let .objectMapping(_, response),
+             let .statusCode(response):
+            return response
+
+        case let .underlying (_, response):
```

Comment:
This could be together with the cases above?

Reply:
underlying's Error is a `Swift.Error` while users are `MoyaError` so the compiler raise a type mismatch.